core/src/main/java/de/jplag/Submission.java

```diff
@@ -265,4 +269,29 @@ private static File createErrorDirectory(String... subdirectoryNames) {
         }
         return true;
     }
+
+    void normalize() {
+        List<Integer> originalOrder = getOrder(tokenList);
+        TokenStringNormalizer.normalize(tokenList);
+        List<Integer> normalizedOrder = getOrder(tokenList);
+
+        logger.debug("original line order: {}", originalOrder);
+        logger.debug("line order after normalization: {}", normalizedOrder);
+        Set<Integer> normalizedSet = new HashSet<>(normalizedOrder);
+        List<Integer> removed = originalOrder.stream().filter(l -> !normalizedSet.contains(l)).toList();
+        logger.debug("removed {} line(s): {}", removed.size(), removed);
+    }
+
+    private List<Integer> getOrder(List<Token> tokenList) {
+        List<Integer> order = new ArrayList<>(tokenList.size());  // a little too big
```

Comment:
Does this increase the performance ? Otherwise it's not needed to define the size

Reply:
I haven't tested it but I would assume so.